### ISSUE TITLE
Screen reader does not scroll a list when using `Localizations.override`

### ISSUE THREAD

**Issue by jiahaog** [flutter/flutter#178432]
Internal Bug: b/458784407. Copying this from the bug.

### Steps to reproduce

 - The `CustomScrollView` below has one item (`Column`) consisting of 50 `ListTile`. 
- If we wrap the entire `Column` with `Localizations.override`, then the screen reader cannot reach to the last `ListTile`
- If we wrap the individual `ListTile` with `Localizations.override`, or delete the `Localizations.override`, then the screen reader can navigate freely.

The following code can run on https://dartpad.dev/, and this can be reproduced with (1) Android Talkback + Chrome and (2) MacOS + Chrome + VoiceOver.

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/semantics.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(const MaterialApp(home: LocalizationBugRepro()));
  SemanticsBinding.instance.ensureSemantics();
}

class LocalizationBugRepro extends StatelessWidget {
  const LocalizationBugRepro({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: CustomScrollView(
        slivers: [
          SliverToBoxAdapter(
            child: Localizations.override(
              context: context,
              locale: const Locale('en', 'US'),
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: List.generate(
                  50,
                  (index) => ListTile(
                    title: Text(
                      "item $index",
                      style: const TextStyle(fontWeight: FontWeight.w600),
                    ),
                    subtitle: Text("some subtitle"),
                    trailing: const Icon(Icons.chevron_right),
                  ),
                ),
              ),
            ),
          ),
        ],
      ),
    );
  }
}
```

</details>

### Flutter Doctor output

Can be reproduced on dartpad with on Dart SDK 3.9.2 and Flutter SDK 3.35.7

**Comment by tirth-patel-nc** [comment]
@jiahaog Thanks for the report. Seeing the same behaviour with latest SDK versions.

```
stable : 3.38.0
master : 3.39.0-1.0.pre-93
```

### Original Code

https://github.com/user-attachments/assets/6bc61ea7-aff0-4694-8895-128b2bb8e02f

### Wrapping individual ListTile with Localizations.override

https://github.com/user-attachments/assets/2b9c848a-4ce0-458f-9d2c-8fc17162f367

**Comment by Renzo-Olivares** [comment]
Closing this issue as a duplicate of https://github.com/flutter/flutter/issues/173097.

The root cause similar to #173097 is that the semantics geometry update is short-circuiting below https://github.com/flutter/flutter/blob/610c06fae6a41dc4d8db0cb880e464c73319399d/packages/flutter/lib/src/rendering/object.dart#L5833

This short-circuit logic is meant to optimize for scrolling.
